WritableMultipart.create method does not work properly

Both `WriteableMultiPart#create` methods are wrapping passed `WriteableBodyPart` instance into the new WriteableBodyParts.
```java
public static WriteableMultiPart create(WriteableBodyPart... entities) {
    Builder builder = builder();
    for (WriteableBodyPart entity : entities) {
        builder.bodyPart(WriteableBodyPart.create(entity));
    }
    return builder.build();
}
```
This causes the following exception:
```
java.lang.IllegalStateException: Transformation failed!
        ... < some long stack trace >
Caused by: java.lang.IllegalStateException: No writer found for type: class io.helidon.media.multipart.WriteableBodyPart. This usually occurs when the appropriate MediaSupport has not been added.
        at io.helidon.media.common.MessageBodyWriterContext.marshall(MessageBodyWriterContext.java:376)
        ... 42 more
```

--- a/media/multipart/src/main/java/io/helidon/media/multipart/WriteableMultiPart.java
+++ b/media/multipart/src/main/java/io/helidon/media/multipart/WriteableMultiPart.java
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2020, 2022 Oracle and/or its affiliates.
+ * Copyright (c) 2020, 2023 Oracle and/or its affiliates.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.

--- a/media/multipart/src/main/java/io/helidon/media/multipart/WriteableMultiPart.java
+++ b/media/multipart/src/main/java/io/helidon/media/multipart/WriteableMultiPart.java
@@ -50,7 +50,7 @@ public class WriteableMultiPart {
     public static WriteableMultiPart create(WriteableBodyPart... entities) {
         Builder builder = builder();
         for (WriteableBodyPart entity : entities) {
-            builder.bodyPart(WriteableBodyPart.create(entity));
+            builder.bodyPart(entity);
         }
         return builder.build();
     }
@@ -65,7 +65,7 @@ public class WriteableMultiPart {
     public static WriteableMultiPart create(Collection<WriteableBodyPart> entities) {
         Builder builder = builder();
         for (WriteableBodyPart entity : entities) {
-            builder.bodyPart(WriteableBodyPart.create(entity));
+            builder.bodyPart(entity);
         }
         return builder.build();
     }

--- a/media/multipart/src/test/java/io/helidon/media/multipart/WritableMultiPartTest.java
+++ b/media/multipart/src/test/java/io/helidon/media/multipart/WritableMultiPartTest.java
@@ -1,0 +1,46 @@
+/*
+ * Copyright (c) 2023 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.helidon.media.multipart;
+
+import java.util.List;
+
+import org.junit.jupiter.api.Test;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.CoreMatchers.is;
+
+/**
+ * Test {@link WriteableMultiPart}.
+ */
+public class WritableMultiPartTest {
+
+    @Test
+    public void testWritableMultiPartCreation() {
+        WriteableBodyPart bodyPart = WriteableBodyPart.create("MyStringEntity");
+        List<WriteableBodyPart> expectedBodyParts = List.of(bodyPart);
+        WriteableMultiPart multiPart = WriteableMultiPart.create(bodyPart);
+        assertThat(multiPart.bodyParts(), is(expectedBodyParts));
+
+        multiPart = WriteableMultiPart.create(List.of(bodyPart));
+        assertThat(multiPart.bodyParts(), is(expectedBodyParts));
+
+        multiPart = WriteableMultiPart.builder()
+                .bodyPart(bodyPart)
+                .build();
+        assertThat(multiPart.bodyParts(), is(expectedBodyParts));
+    }
+
+}

